wordpress-mobile/WordPress-iOS
# Blogging Prompts: Add Core Data model and adjust service

Refs #18375

So... it's a pretty big chunk of changes. But most of it is Core Data, I promise! 😆 I originally intended to split the PR between adding _just_ the Core Data model, and adding the changes to the `BloggingPromptsService` — but there's a naming conflict between the temporary model and the Core Data model so I decided to just fix it in one go.

A couple of notes about the changes:

- Considering the size of changes and the fact that this touches Core Data, I'm thinking to target this for 20.0. I'm just hoping that this won't turn into a gigantic conflict after the release branch is cut 😬 .
- Say goodbye to the temporary model. `BloggingPrompt` is now a Core Data model. 
  - ⚠️ Note: I've added a `siteID` field to `BloggingPrompt` since the user can access multiple sites — but there shouldn't be any impact on the UI side since this has been handled internally in `BloggingPromptsService`. Just make sure that you're always getting the prompts from the service, or remember to add `siteID` when fetching the model directly.
  - Added a convenience method `promptAttribution` that returns `BloggingPromptsAttribution`. 
- To remove the references to `examplePrompt`, I've replaced the calls with the `prompt` instance variable in `DashboardPromptCardCell`, and `CreateButtonCoordinator`.
- As for the changes on `BloggingPromptsService`, it now properly creates new objects and/or updates existing ones.
  - Additionally, I've added a convenience `localTodaysPrompt` method which synchronously fetches today's prompt from Core Data. I'm sure there's a use case for this somewhere. 
- Added some more unit tests.

## To Test

### Unit tests ⚙️ 

Make sure that the unit tests are passing.

### Manual tests 👀 

- Enable the blogging prompts feature flag.
- Verify that the prompt contents are displayed correctly in the dashboard card, FAB, and the post draft (after tapping the Answer Prompt).

Description | Screenshot
--|--
Dashboard card | ![dashboard-card](https://user-images.githubusercontent.com/1299411/168312066-99ffdcde-fba7-4b19-bfce-80fd222acffb.png)
Floating action button | ![create-post](https://user-images.githubusercontent.com/1299411/168312048-4a25fd3e-688f-40e8-b50e-8d334cfa96a7.png)
Post draft | ![post-draft](https://user-images.githubusercontent.com/1299411/168312059-f9149d31-8280-4c66-9a33-fe631c2a2d96.png)

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is not released yet.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is not released yet.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for object mapping and date filtering.

PR submission checklist:
